sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/InsertWindowGroupLimitSuite.scala

```diff
@@ -0,0 +1,308 @@
+
+      val correctAnswer1 =
+        testRelation
+          .windowGroupLimit(a :: Nil, c.desc :: Nil, function, 2)
+          .select(a, b, c,
+            windowExpr(function,
+              windowSpec(a :: Nil, c.desc :: Nil, windowFrame)).as("rn"))
+          .where(condition && b > 0)
+
+      comparePlans(
+        Optimize.execute(originalQuery1.analyze),
+        WithoutOptimize.execute(correctAnswer1.analyze))
+
+      val originalQuery2 =
+        testRelation
+          .select(a, b, c,
+            windowExpr(function,
+              windowSpec(Nil, c.desc :: Nil, windowFrame)).as("rn"))
+          .where(condition)
+
+      val correctAnswer2 =
+        testRelation
+          .windowGroupLimit(Nil, c.desc :: Nil, function, 2)
+          .select(a, b, c,
+            windowExpr(function,
+              windowSpec(Nil, c.desc :: Nil, windowFrame)).as("rn"))
+          .where(condition)
+
+      comparePlans(
+        Optimize.execute(originalQuery2.analyze),
+        WithoutOptimize.execute(correctAnswer2.analyze))
+
+      val originalQuery3 =
+        testRelation
+          .select(a, b, c,
+            windowExpr(function,
+              windowSpec(Nil, c.desc :: Nil, windowFrame)).as("rn"))
+          .where(condition && b > 0)
+
+      val correctAnswer3 =
+        testRelation
+          .windowGroupLimit(Nil, c.desc :: Nil, function, 2)
+          .select(a, b, c,
+            windowExpr(function,
+              windowSpec(Nil, c.desc :: Nil, windowFrame)).as("rn"))
+          .where(condition && b > 0)
+
+      comparePlans(
+        Optimize.execute(originalQuery3.analyze),
+        WithoutOptimize.execute(correctAnswer3.analyze))
+    }
+  }
+
+  test("Unsupported conditions") {
+    for (condition <- unsupportedConditions; function <- rankLikeFunctions) {
+      val originalQuery =
+        testRelation
+          .select(a, b, c,
+            windowExpr(function,
+              windowSpec(a :: Nil, c.desc :: Nil, windowFrame)).as("rn"))
+          .where(condition)
+
+      comparePlans(
+        Optimize.execute(originalQuery.analyze),
+        WithoutOptimize.execute(originalQuery.analyze))
+    }
+  }
+
+  test("Unsupported window functions") {
+    for (condition <- supportedConditions; function <- unsupportedFunctions) {
+      val originalQuery =
+        testRelation
+          .select(a, b, c,
+            windowExpr(function,
+              windowSpec(a :: Nil, c.desc :: Nil, windowFrame)).as("rn"))
+          .where(condition)
+
+      comparePlans(
+        Optimize.execute(originalQuery.analyze),
+        WithoutOptimize.execute(originalQuery.analyze))
+    }
+  }
+
+  test("Insert window group limit node for top-k computation: Empty partitionSpec") {
+    rankLikeFunctions.foreach { function =>
+      val originalQuery =
+        testRelation
+          .select(a, b, c,
+            windowExpr(function,
+              windowSpec(Nil, c.desc :: Nil, windowFrame)).as("rn"))
```

Comment:
seems it's already tested: https://github.com/apache/spark/pull/38799/files#diff-3152a64564129e341027e179be3483e40972174b650b1228a0ac9755ab92adb3R134

Reply:
Thank you for you reminder.